File: finance/src/main/java/mg/finance/entity/Currency.java
package mg.finance.entity;

import lombok.*;

import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.Set;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@Table(name="mg_currency")
@NoArgsConstructor
@AllArgsConstructor
public class Currency {

    @Id
    @GeneratedValue
    private UUID uuid;
    private long systemId;
    private OffsetDateTime date;
    private String abbreviation;
    private double scale;
    private String name;
    private double rate;
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @OneToMany(mappedBy = "currency", orphanRemoval = true, cascade = CascadeType.ALL)
    private Set<CurrencyStatistics> statistics;
}

File: webui/src/main/webapp/src/models/currency/currency.js
export class Currency {

    /**
     * @param {Object} config
     * @param {String} config.uuid
     * @param {String} config.date
     * @param {String} config.abbreviation
     * @param {String} config.name
     * @param {Number} config.systemId
     * @param {Number} config.scale
     * @param {Number} config.rate
     *
     */
    constructor(config) {
        config = config || {};
        this.uuid = config.uuid;
        this.date = config.date;
        this.abbreviation = config.abbreviation;
        this.name = config.name;
        this.systemId = config.systemId;
        this.scale = config.scale;
        this.rate = config.rate;
    }
}